ElementX/Sources/Screens/ComposerToolbar/View/VoiceMessageRecordingButton.swift

```diff
@@ -17,79 +17,60 @@
+            switch mode {
+            case .standard:
                 startRecording?()
-            } else {
-                if let recordingStartTime, Date.now.timeIntervalSince(recordingStartTime) < minimumRecordingDuration {
-                    withElementAnimation {
-                        showTooltip = true
-                    }
-                    notificationFeedbackGenerator.notificationOccurred(.error)
-                    stopRecording?(false)
-                } else {
-                    impactFeedbackGenerator.impactOccurred()
-                    stopRecording?(true)
-                }
-            }
-        }
-        .overlay(alignment: .bottomTrailing) {
-            if showTooltip {
-                tooltipView
-                    .offset(y: -frame.height - tooltipPointerHeight)
+            case .recording:
+                stopRecording?()
             }
-        }
+        } label: { }
+            .accessibilityLabel(L10n.a11yVoiceMessageRecord)
+            .buttonStyle(VoiceMessageRecordingButtonStyle(mode: mode))
     }
+}
+
+private struct VoiceMessageRecordingButtonStyle: ButtonStyle {
+    let mode: VoiceMessageRecordingButtonMode
     
-    private var tooltipView: some View {
-        VoiceMessageRecordingButtonTooltipView(text: L10n.screenRoomVoiceMessageTooltip,
-                                               pointerHeight: tooltipPointerHeight,
-                                               pointerLocation: frame.midX,
-                                               pointerLocationCoordinateSpace: .global)
-            .allowsHitTesting(false)
-            .fixedSize()
-            .onAppear {
-                DispatchQueue.main.asyncAfter(deadline: .now() + tooltipDuration) {
-                    withElementAnimation {
-                        showTooltip = false
-                    }
-                }
+    func makeBody(configuration: Configuration) -> some View {
+        Group {
+            switch mode {
+            case .standard:
+                CompoundIcon(configuration.isPressed ? \.micOnSolid : \.micOnOutline)
+                    .foregroundColor(.compound.iconSecondary)
+                    .padding(EdgeInsets(top: 6, leading: 6, bottom: 6, trailing: 6))
+            case .recording:
+                Image(systemName: "stop.circle.fill")
+                    .resizable()
+                    .frame(width: 34, height: 34)
             }
+        }
     }
 }
 
 struct VoiceMessageRecordingButton_Previews: PreviewProvider, TestablePreview {
     static var previews: some View {
-        VoiceMessageRecordingButton()
-            .fixedSize(horizontal: true, vertical: true)
+        HStack {
+            VoiceMessageRecordingButton(mode: .standard)
+                .fixedSize(horizontal: true, vertical: true)
+            
+            VoiceMessageRecordingButton(mode: .recording)
+                .fixedSize(horizontal: true, vertical: true)
+        }
```

Comment:
```suggestion
        HStack {
            VoiceMessageRecordingButton(mode: .standard)
            VoiceMessageRecordingButton(mode: .recording)
        }
```
I'm not sure why you use `.fixedSize` here.
Is there a specific reason? 
The preview looks the same to me if I delete it.

Reply:
Removed.